recipes/mysql-connector-cpp/all/test_package/conanfile.py

```diff
@@ -0,0 +1,32 @@
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
+
+import os
+
+class TestPackageConan(ConanFile):
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        self.with_jdbc = self.dependencies["mysql-connector-cpp"].options.with_jdbc
+
+    def layout(self):
+        cmake_layout(self)
+
+    def build(self):
+        cmake = CMake(self)
+        if self.with_jdbc:
+            cmake.configure({"JDBC_ENABLED": "ON"})
+        else:
+            cmake.configure({"JDBC_ENABLED": "OFF"})
+        cmake.build()
```

Comment:
```suggestion
        tc = CMakeToolchain(self)
        tc.variables["JDBC_ENABLED"] = self.dependencies["self.tested_reference_str].options.with_jdbc
        tc.generate()

    def layout(self):
        cmake_layout(self)

    def build(self):
        cmake = CMake(self)
        cmake.build()
```

Please, use only CMakeToolchain. Also, need to update import and generators attribute.

Reply:
@uilianries thanks for your comment.

if i use CmakeToolcahin as you mention in the test i get this error

```
ERROR: mysql-connector-cpp/8.3.0 (test package): Error in generate() method, line 18
        tc.generate()
        ConanException: CMakeToolchain is declared in the generators attribute, but was instantiated in the generate() method too. It should only be present in one of them.
```